common/api/src/main/java/com/bazaarvoice/emodb/common/api/Names.java

```diff
@@ -28,4 +36,16 @@ public static boolean isLegalTableName(String table) {
                 !(table.charAt(0) == '.' && (".".equals(table) || "..".equals(table))) &&
                 TABLE_NAME_ALLOWED.matchesAllOf(table);
     }
+
+    /**
+     * Role names mostly follow the same conventions as table names except, since they are only used internally, are
+     * slightly more permissive, such as allowing capital letters.  Although "_" is a valid part of a role name it
+     * can't be used as the first character.
+     */
+    public static boolean isLegalRoleName(String role) {
+        return role != null &&
+                role.length() > 0 && role.length() <= 255 &&
+                !role.startsWith("_") &&
+                ROLE_NAME_ALLOWED.matchesAllOf(role);
```

Comment:
Minor: why are we just restricting role names not to have _ as first character? why not just not have _ at all OR no such restriction on _ character? 
I see no difference between _/_role1  and _/role_1
Also, I see that we using the same method for validating group name (Line 69 in TableRoleManager)(should we generalize the method name or split them if we have separate requirements :?). you think we shouldn't allow _group1/role1 as we may have _/role1 :?